app/code/Magento/Customer/Console/Command/UpgradeHashAlgorithmCommand.php

```diff
@@ -65,19 +91,31 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $customerCollection = $this->collection->getItems();
         /** @var $customer Customer */
         foreach ($customerCollection as $customer) {
-            $customer->load($customer->getId());
             if (!$this->encryptor->validateHashVersion($customer->getPasswordHash())) {
                 list($hash, $salt, $version) = explode(Encryptor::DELIMITER, $customer->getPasswordHash(), 3);
                 $version .= Encryptor::DELIMITER . $this->encryptor->getLatestHashVersion();
                 $hash = $this->encryptor->getHash($hash, $salt, $this->encryptor->getLatestHashVersion());
                 list($hash, $salt) = explode(Encryptor::DELIMITER, $hash, 3);
                 $hash = implode(Encryptor::DELIMITER, [$hash, $salt, $version]);
-                $customer->setPasswordHash($hash);
-                $customer->save();
+                $customerDataObject = $this->getCustomerDataObject($customer->getId());
+                $this->customerRepository->save($customerDataObject, $hash);
                 $output->write(".");
             }
         }
         $output->writeln(".");
         $output->writeln("<info>Finished</info>");
     }
+
+    /**
+     * Get customer data object
+     *
+     * @param int $customerId
+     * @return CustomerInterface
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
+     */
+    private function getCustomerDataObject(int $customerId)
```

Comment:
What's the purpose of this method? It would be better inline it

Reply:
> It would be better inline it

**implemented**